### ISSUE TITLE
Allow to edit existing questions, surveys, and deployments

### ISSUE THREAD

**Issue by jygh98** [erdl/survey_admin#33]
Currently survey admin only allows for users to add to the database, but to make any changes or deletion will require you to do so via the database.  For this, it only makes sense that all insertion and edits should be done from survey admin to prevent the risk doing undesired along with consistency purposes of having survey admin to do it.

**Comment by jygh98** [comment]
@carlosparadis

**Comment by carlosparadis** [comment]
:v some notes 

> Currently survey admin only allows for users to add to the database


To add what?

> , but to make any changes or deletion will require you to do so via the database.


To make changes and deletions to what?

>  For this, it only makes sense that all insertion and edits should be done from survey admin **to prevent the risk doing undesired along with consistency purposes** of having survey admin to do it.

The english looks strange in the bold part!

Try to list the description with all possible actions eileen and jim can do when using survey_admin. Then try to see how many we cover, how many the new page you add cover, and then what is missing.

**Comment by jygh98** [comment]
So as of now these are the possible actions that eileen and jim are able to do on survey admin atm:
1. view questions 
2. add questions 
3. view surveys
4. create surveys
5. view deployments
6. create deployments
7. edit deployment *(currently we are only able to edit the survey that is being selected)

Ideally we would like survey admin to be able to cover all possible features including
1. editing questions
    a. changing question text, description, and options selected
2. editing surveys
    a. changing survey name, description, and questions selected
3. enable the edit functionality for deployments to do more
    a. be able to change url, building selected, and kiosk view
4. being able to remove questions, surveys, and deployments

Also in the OAuth PR i included a page that will users to add people to the survey_admin, meaning jim and eileen can add anyone so that they can login through oauth.

**Comment by jygh98** [comment]
Survey_admin response to issue [#37](https://github.com/erdl/survey_display/issues/37)

Currently we are able to add and edit any questions, surveys, and deployments through survey_admin.  However due to the foreign key constraints from response table, deleting a survey from the database should be avoided.  

For this, we decided to implement an "active" state for every deployment that can be changed through survey admin.  When a new deployment is created, by default it would be "inactive" in the case we would like to hold off for deployment for a specific time/date.  When the deployment is ready to be viewed on survey_display, you can go and edit a deployment to deploy it.

**Comment by carlosparadis** [comment]
@ryantanaka I am rotating this issue to you. Consider it on hold until we sort the priorities tomorrow.

**Comment by carlosparadis** [comment]
@ryantanaka with the last batch of commits merged in, is this the next pending item or is it blocked by the SSL Certificate?

**Comment by carlosparadis** [comment]
@ryantanaka 

Eileen told me we will very likely be deploying the surveys at the start of the semester, about 3 weeks from now.

Did the SSL Certificate got fully integrated? And is this issue done? We will likely want to have this wrap up before then and make some tests on all the functionality.

**Comment by ryantanaka** [comment]
SSL certificate was fully integrated. HTTPS is used for all our web apps and if someone goes to any of our urls using HTTP, it automatically redirects to HTTPS (thanks to Justin!) 

Notes: @carlosparadis 

Got it this will be done soon. So we decided that what should be editable is the questions that are in a survey. We won't change the questions themselves because that ruins our responses when we look back in time to see what responses correspond with what questions... 

If someone wants to change a survey, we agreed also that I would put some kind of reminder/mechanism in place so that any user doesn't carelessly change a survey question...

**Comment by ryantanaka** [comment]
Hey @carlosparadis,

Here is an update of what I have based on what we discussed. Surveys can be edited (questions removed and added). 

![screenshot from 2018-08-13 15-24-38](https://user-images.githubusercontent.com/19616012/44066632-07833c8e-9f0d-11e8-86bb-febb24b14f3d.png)

The form looks like this when you click on "Edit" from the image above:

![screenshot from 2018-08-13 15-34-04](https://user-images.githubusercontent.com/19616012/44066856-5538d550-9f0e-11e8-9342-006f1a94f7d2.png)

When the page loads, questions that are already in use by the survey are pre-checked and colored red. If new questions are selected and submit is clicked, the form will change accordingly when it is reloaded. 

No two step verification type thing yet, but I'll try add that in.

**Comment by carlosparadis** [comment]
Looks great! May want to change the color of the text from gray on the one on the grey background!

**Comment by ryantanaka** [comment]
Great! I'll do that.

**Comment by ryantanaka** [comment]
@carlosparadis 

Hey so currently our "Create Survey" form looks like this:

![screenshot from 2018-08-15 11-05-52](https://user-images.githubusercontent.com/19616012/44173307-3f67f800-a07b-11e8-9c7d-c6ce1cc77a85.png)

I was going to use the same type of form for the "Edit Survey", but I wanted to try  the multi-checkbox one because it's a little simpler for the user to use. For example, in order to select multiple questions via the "Create Survey" form, a user has to `ctrl-click` multiple items. This is implied and not so intuitive. For consistency, I should use the same type of form when we need to edit a question. 

I'm thinking that I should switch the `Which questions should be a part of this survey?` portion of the "Create Survey" form to be the same as I've done it for "Edit Survey" so everything looks coherent. 

Any thoughts on this?

**Comment by ryantanaka** [comment]
Actually, let's table that since it's not like the "Edit Survey" feature will be used often. That can be added later on by whoever. I'm gonna finish making sure everything works the way we want it as is and then submit a PR for this issue.
